src/core/rules.rkt

```diff
@@ -225,6 +235,13 @@
                  [difference-of-sqr--1 (+ (* a a) -1) (* (+ a 1) (- a 1))]
                  [pow-sqr (* (pow a b) (pow a b)) (pow a (* 2 b))])
 
+(define-ruleset* difference-of-squares-canonicalize-revv
+                 (polynomials simplify sound)
+                 #:type ([a real] [b real])
+                 [difference-of-sqr-1-rev (* (+ a 1) (- a 1)) (- (* a a) 1)]
+                 [difference-of-sqr--1-rev (* (+ a 1) (- a 1)) (+ (* a a) -1)]
+                 [difference-of-squares-rev (* (+ a b) (- a b)) (- (* a a) (* b b))]
+                 [pow-sqr-rev (pow a (* 2 b)) (* (pow a b) (pow a b))])
```

Comment:
Is the `pow-sqr-rev` rule true? Consider `(pow -1 (* 2 1/2)) = (pow -1 1) = -1` but `(* (pow -1 1/2) (pow -1 1/2))` is illegal because `(pow -1 1/2)` does not exist.

Reply:
The rule is technically true considering complex numbers, does herbie not deal with complex numbers as I thought some of bnechmarks used complex?

Reply:
Real numbers only. Complex numbers are even worse.